src/main/java/org/folio/note/NoteRepositoryImpl.java

```diff
@@ -0,0 +1,205 @@
+package org.folio.note;
+
+import static io.vertx.core.Future.failedFuture;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+
+import org.folio.db.model.NoteView;
+import org.folio.rest.jaxrs.model.Note;
+import org.folio.rest.jaxrs.model.NoteCollection;
+import org.folio.rest.persist.PgExceptionUtil;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.Criteria.Limit;
+import org.folio.rest.persist.Criteria.Offset;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.persist.interfaces.Results;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSON;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSONException;
+import org.z3950.zing.cql.cql2pgjson.FieldException;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+
+@Component
+public class NoteRepositoryImpl implements NoteRepository {
```

Comment:
regarding method naming.. 
Your primary entity is Note. It is assumed that methods do something with a note or several notes, so the method names could omit "Note(s)" without losing the meaning: 
* save instead of saveNote;
* delete instead of deleteNote
* and so on
also I'd suggest to start method names that retrieve entities from "find"

If we follow the same naming, a general repository interface can be extracted later on to get something similar to Spring CRUD repository:
```
public interface CrudRepository<T, ID extends Serializable>
  extends Repository<T, ID> {
 
    <S extends T> S save(S entity);
 
    T findOne(ID primaryKey);
 
    Iterable<T> findAll();
 
    Long count();
 
    void delete(T entity);
 
    boolean exists(ID primaryKey);
}
```